core/src/main/java/com/klaytn/caver/account/AccountKeyRoleBased.java

```diff
@@ -0,0 +1,202 @@
+     * TxTypeAccountUpdate transaction is validated using RoleTransaction key.
+     *
+     * Third Key : roleFeePayer
+     * If this account wants to send tx fee instead of the sender, the transaction should be signed by this key.
+     * If this key is not present in the account, a fee-delegated transaction is validated using RoleTransaction key.
+     */
+    private List<IAccountKey> accountKeys;
+
+    /**
+     * Creates an AccountKeyRoleBased instance.
+     * @param accountKeys List of a AccountKey implements IAccountKey interface
+     */
+    public AccountKeyRoleBased(List<IAccountKey> accountKeys) {
+        setAccountKeys(accountKeys);
+    }
+
+    /**
+     * Decodes a RLP-encoded AccountKeyRoleBased string
+     * @param rlpEncodedKey RLP-encoded AccountKeyRoleBased string.
+     * @return AccountKeyRoleBased
+     */
+    public static AccountKeyRoleBased decode(String rlpEncodedKey) {
+        return decode(Numeric.hexStringToByteArray(rlpEncodedKey));
+    }
+
+    /**
+     * Decodes a RLP-encoded AccountKeyRoleBased byte array
+     * @param rlpEncodedKey RLP-encoded AccountKeyRoleBased byte array
+     * @return AccountKeyRoleBased
+     */
+    public static AccountKeyRoleBased decode(byte[] rlpEncodedKey) {
+        byte type = Numeric.hexStringToByteArray(getType())[0];
+        if(rlpEncodedKey[0] != type) {
+            throw new IllegalArgumentException("Invalid RLP-encoded AccountKeyRoleBased Tag");
+        }
+
+        List<IAccountKey> accountKeys = new ArrayList<>();
+        //remove Tag
+        byte[] encodedKey = Arrays.copyOfRange(rlpEncodedKey, 1, rlpEncodedKey.length);
+
+        RlpList rlpList = RlpDecoder.decode(encodedKey);
+        List<RlpType> values = ((RlpList) rlpList.getValues().get(0)).getValues();
+
+        for (RlpType value : values) {
+            accountKeys.add(AccountKeyDecoder.decode(((RlpString) value).asString()));
+        }
+        return new AccountKeyRoleBased(accountKeys);
+    }
+
+    /**
+     * Creates an AccountKeyRoleBased give params.
+     * @param pubArray An List contains public key string array.
+     * @param options An List contains WeightedMultiSigOptions
+     * @return AccountKeyRoleBased
+     */
+    public static AccountKeyRoleBased fromRoleBasedPublicKeysAndOptions(List<String[]> pubArray, List<WeightedMultiSigOptions> options) {
+        List<IAccountKey> accountKeys = new ArrayList<>();
+
+        /*
+        * pubArray must have three item.
+        * each item can have
+        * {String, String...}, {}
+        *
+        * options must have to three item.
+        * Valid WeightedMultiSigOption or Empty WeightedMultiSigOption
+        *  - Valid WeightedMultiSigOption : make AccountKeyWeightedMultiSig instance
+        *  - Empty WeightedMultiSigOption : make AccountKeyPublicKey instance
+        * */
+        if(pubArray.size() != 3 && options.size() != 3) {
```

Comment:
The role can be added in future, so if possible implement role with enum(or something else) will be better.